components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/OauthTokenIssuerImpl.java

```diff
@@ -19,34 +19,79 @@
 
 package org.wso2.carbon.identity.oauth2.token;
 
+import org.apache.axiom.util.base64.Base64Utils;
+import org.apache.commons.io.Charsets;
 import org.apache.oltu.oauth2.as.issuer.OAuthIssuer;
 import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
 import org.wso2.carbon.identity.oauth.config.OAuthServerConfiguration;
 import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
 
 public class OauthTokenIssuerImpl implements OauthTokenIssuer {
 
+
     private OAuthIssuer oAuthIssuerImpl = OAuthServerConfiguration.getInstance()
             .getOAuthTokenGenerator();
 
     public String accessToken(OAuthTokenReqMessageContext tokReqMsgCtx) throws OAuthSystemException {
-        return oAuthIssuerImpl.accessToken();
+        String tokenBindingId = OAuth2Util.checkTokenBindingHeader(tokReqMsgCtx,
+                OAuthConstants.HTTP_TB_REFERRED_HEADER_NAME);
+        String accesstoken = oAuthIssuerImpl.accessToken();
+        if (!tokenBindingId.isEmpty()) {
+            accesstoken = OAuth2Util.hashTB(tokenBindingId) + ";" + accesstoken;
+            accesstoken = baseEncodeTB(accesstoken);
```

Comment:
Change this method as base64Encode please. You are not doing anything TB specific there.

Reply:
okay